### PR TITLE
Empty State Guidance for Dashboard

### PR DESCRIPTION
**Pull request by Fayroth**
## Changes
- Added welcoming message and guidance when the dashboard has no widgets
- Replaced standard FAB with a user-friendly "Add Your First Widget" button in empty state
- Added comprehensive test coverage for all empty state scenarios

## Implementation Details
- Empty state displays when `widgetList` is empty
- Conditional rendering between empty state and widget list based on list size
- FAB is only shown when widgets exist
- Empty state uses Material Design 3 components and styling

## Testing
Added test coverage for:
- Initial empty state display and UI elements
- Adding first widget through empty state button
- State transitions (empty ↔ populated)
- Dialog interactions from empty state
- Widget removal leading to empty state

## Screenshots
<div style="display: flex; gap: 10px;">
    
    
    
</div>

## Additional Notes
- Ensures consistent UX between empty and populated states
- Improves first-time user experience with clear guidance
- Maintains existing widget functionality when populated

### CODE REVIEW

**Comment by Fayroth** [comment]
I need to wait for the merge of the PR refactoring the end2end tests and then I will be able to fix the issues in tests.

File: app/src/main/java/com/github/se/eduverse/ui/dashboard/DashboardScreen.kt [hunk 1]
```diff
@@ -167,166 +174,214 @@ private fun ReorderableWidgetList(
...
+                          IconButton(
+                              onClick = { onDelete(item.widgetId) },
+                              modifier = Modifier.size(24.dp).testTag("delete_icon")) {
+                                Icon(
+                                    Icons.Default.Close,
+                                    contentDescription = "Delete",
+                                    tint = MaterialTheme.colorScheme.error)
+                              }
+                        }
+                  }
```
Comment (ismailIBNMAJAH) [inline app/src/main/java/com/github/se/eduverse/ui/dashboard/DashboardScreen.kt:380]: lgtm. It worked well before and the logic seems similar to me.

File: app/src/androidTest/java/com/github/se/eduverse/ui/DashboardScreenTest.kt [hunk 2]
```diff
@@ -805,15 +805,140 @@ class DashboardScreenUiTest {
...
+    composeTestRule.onNodeWithTag("add_common_widget_button").assertIsDisplayed()
+
+    // Click cancel
+    composeTestRule.onNodeWithText("Cancel").performClick()
+    composeTestRule.waitForIdle()
+
+    // Verify back to empty state
+    composeTestRule.onNodeWithTag("empty_dashboard_message").assertIsDisplayed()
+  }
+
```
Comment (ismailIBNMAJAH) [inline app/src/androidTest/java/com/github/se/eduverse/ui/DashboardScreenTest.kt:926]: The tests are well-implemented and cover key scenarios effectively, including the initial display of the empty state, transitioning to a widget list after adding a widget, and returning to the empty state after deleting all widgets. They also validate user interactions such as opening the add widget dialog and ensuring proper navigation and UI updates. Great attention to detail!

File: app/src/androidTest/java/com/github/se/eduverse/E2E/E2ETests.kt [hunk 3]
```diff
@@ -510,24 +520,34 @@ class PomodoroTimerE2ETest {
...
     composeTestRule.apply {
       waitForIdle()
 
-      // 1. Add Timer Widget
-      onNodeWithTag("add_widget_button").performClick()
+      // 1. Verify Empty State
+      onNodeWithTag("empty_dashboard_message").assertIsDisplayed()
+      onNodeWithTag("empty_state_add_button").assertIsDisplayed()
+      onNodeWithTag("add_widget_button").assertDoesNotExist()
+
```
Comment (ismailIBNMAJAH) [inline app/src/androidTest/java/com/github/se/eduverse/E2E/E2ETests.kt:535]: the e2e test has been nicely updated as a result of the changes, well done!

**Review by ismailIBNMAJAH — APPROVED** [review]
Great implementation! The changes are well-structured, the tests effectively cover significant scenarios, and the end-to-end tests have been properly updated to reflect the modifications. Well done!

**Pull request merged**